ChelseasFieldJournal/src/GUI/GuiDialogToolJournal.cs

```diff
@@ -0,0 +1,202 @@
+        private object currentRightPage;
+
+        public GuiDialogToolJournal(string DialogTitle, EntityPlayer playerIn, ICoreClientAPI capi) : base(DialogTitle, capi)
+        {
+            this.SetupDialogJournal();
+            if (leftPageNumber < 1) { leftPageNumber = 1; }
+            if (rightPageNumber < 1) { rightPageNumber = 1; }
+            //InitializeJournalEntries();
+            //if (ChapterPage < 1) { ChapterPage = 1; }
+        }
+
+        private void SetupDialogJournal()
+        {
+            IGuiAPI gui = this.capi.Gui;
+            string dialogId = "gui-dialog-journal";
+            ElementBounds dialogBounds = ElementStdBounds.AutosizedMainDialog.WithAlignment(EnumDialogArea.CenterMiddle);
+            ElementBounds JournalBlankBounds = ElementBounds.Fixed(0.0, 0.0, 1091, 700);
+            ElementBounds TestIconBounds = ElementBounds.Fixed(650.0, 100.0, 256.0, 196.0);
+            ElementBounds ItemStackIconBounds = ElementBounds.Fixed(200.0, 200.0, 256.0, 256.0);
+            ElementBounds childBounds = new ElementBounds().WithSizing(ElementSizing.FitToChildren);
+
+            //TitlePageEntry currentChapterPage = JournalTitleEntries[Chapter];
+            Console.WriteLine("title Entries:" + JournalTitleEntries.Count);
+            Console.WriteLine("page Entries:" + JournalPageEntries.Count);
+            Console.WriteLine("Fauna Entries:" + JournalFaunaEntries.Count);
+            Console.WriteLine("Flora Entries:" + JournalFloraEntries.Count);
+            Console.WriteLine("Cooking Entries:" + JournalCookingEntries.Count);
+            Console.WriteLine("Construct Entries:" + JournalConstructEntries.Count);
+
+                // Array of lists containing all your journal entries
+                var journalChapters = new List<object>[] {
+                JournalPageEntries.Cast<object>().ToList(),
+                JournalOreEntries.Cast<object>().ToList(),
+                JournalFaunaEntries.Cast<object>().ToList(),
+                JournalFloraEntries.Cast<object>().ToList(),
+                JournalCookingEntries.Cast<object>().ToList(),
+                JournalConstructEntries.Cast<object>().ToList()
+            };
+
+                if (Chapter >= 0 && Chapter < journalChapters.Length)
+                {
+                    var Entry = journalChapters[Chapter];
+                    if (ChapterLeftPage < Entry.Count)
+                    {
+                        currentLeftPage = Entry[ChapterLeftPage];
+                        currentRightPage = (ChapterLeftPage + 1 < Entry.Count) ? Entry[ChapterLeftPage + 1] : null;
+                    }
+                    else if (Chapter + 1 < journalChapters.Length)
+                    {
+                        Chapter++;
+                        ChapterLeftPage = 0;
+                        currentLeftPage = Entry[ChapterLeftPage];
+                        currentRightPage = (ChapterLeftPage + 1 < Entry.Count) ? Entry[ChapterLeftPage + 1] : null;
```

Comment:
I would, instead of setting it to null if we can't continue, simply set it to the currentRightPage. This would act as if it's blocking, which would be better than the page becoming a null and possibly cause a null ref somewhere else.